maildir/message.c

```diff
@@ -533,7 +533,7 @@ bool maildir_msg_open_new(struct Mailbox *m, struct Message *msg, const struct E
 {
   int fd;
   char path[PATH_MAX] = { 0 };
-  char suffix[16] = { 0 };
+  char suffix[PATH_MAX] = { 0 };
```

Comment:
My first though was that `path`+`subdir`+`suffix` > `PATH_MAX` but it seems that the resulting `path` is limited to `PATH_MAX` so we should be good.